src/fonduer/parser/models/utils.py

```diff
@@ -15,13 +15,30 @@ def construct_stable_id(
     Contruct a stable ID for a Context given its parent and its character
     offsets relative to the parent.
     """
-    doc_id, _, parent_doc_char_start, _ = split_stable_id(parent_context.stable_id)
+
+    doc_id, _, idx = split_stable_id(parent_context.stable_id)
+
+    # Caption, document, figure, paragraph, section, table
+    if len(idx) == 1:
+        parent_doc_start = idx[0]
+        return f"{doc_id}::{polymorphic_type}:{parent_doc_start}"
+    # Cell
+    elif len(idx) == 3:
```

Comment:
This feels like a magic number to me. Why is this a cell? Is there a way to make this code easier to read than relying on len(idx)?